### PR TITLE
Docs cleanup

### PR DESCRIPTION
**Pull request by kvrhdn**
**What this PR does**:
I went through the docs and cleaned up some broken links / outdated sections. I'll start a code review to discuss the parts I'm less sure about.

**Which issue(s) this PR fixes**:
~~Fixes #<issue number>~~

**Checklist**
Not relevant, only the docs changed.

### CODE REVIEW

File: docs/tempo/website/configuration/_index.md [hunk 1]
```diff
@@ -10,15 +10,14 @@ This section explains the configuration options for Tempo as well as the details
...
+  - [Query Frontend](#query-frontend)
   - [Querier](#querier)
   - [Compactor](#compactor)
   - [Storage](#storage)
   - [Memberlist](#memberlist)
-  - [Compression](#compression)
 
 ## Authentication/Server
-Tempo uses the Weaveworks/common server.  See [here](https://github.com/weaveworks/common/blob/main/server/server.go#L45) for all configuration options.
+Tempo uses the Weaveworks/common server. See [here](https://github.com/weaveworks/common/blob/master/server/server.go#L45) for all configuration options.
```
Comment (kvrhdn) [inline docs/tempo/website/configuration/_index.md:20]: I'm not sure why this had to change from `main` to `master`, but I can't find a `main` branch on https://github.com/weaveworks/common/branches 🤷🏻

File: docs/tempo/website/configuration/manifest.md [hunk 2]
```diff
@@ -18,7 +18,6 @@ go run ./cmd/tempo --storage.trace.backend=local --storage.trace.local.path=/tmp
 
 ```yaml
 target: all
-multitenancy_enabled: true
```
Comment (kvrhdn) [inline docs/tempo/website/configuration/manifest.md:21]: This is the result of running the listed command again. Should updating this segment be part of our release procedure?

**Review by kvrhdn — COMMENTED** [review]
I'm not sure why `example/readme.md` was just an empty file, but I guess we should either remove it or fill it in properly.

File: docs/tempo/website/configuration/_index.md (hunk 1 above)
Comment (dgzlopes) [reply docs/tempo/website/configuration/_index.md:20]: Probably `s/master/main/` (on #655) changed this, and no-one noticed :smile:

Comment (kvrhdn) [reply docs/tempo/website/configuration/_index.md:20]: Ah, makes sense. Another mystery solved 😃

File: docs/tempo/website/configuration/_index.md [hunk 3]
```diff
@@ -10,15 +10,14 @@ This section explains the configuration options for Tempo as well as the details
   - [Authentication/Server](#authenticationserver)
   - [Distributor](#distributor)
   - [Ingester](#ingester)
-  - [Query Frontend](#queryfrontend)
+  - [Query Frontend](#query-frontend)
   - [Querier](#querier)
   - [Compactor](#compactor)
   - [Storage](#storage)
   - [Memberlist](#memberlist)
-  - [Compression](#compression)
```
Comment (kvrhdn) [inline docs/tempo/website/configuration/_index.md:18]: This section was added by https://github.com/grafana/tempo/commit/3b3b9e2c5e93ad59144a8b6c5039fc6f982256a0. But contrary to the other sections, this doesn't refer to a header within this page and should link to a different page (which is listed in the sidebar already).

People should be able to discover the page using the sidebar, so I think we can remove this.

File: docs/tempo/website/configuration/manifest.md (hunk 2 above)
Comment (dgzlopes) [reply docs/tempo/website/configuration/manifest.md:21]: I like that idea! 

If we don't add something like that... most people (at least me:)) will forget to update that section on PRs.

Comment (kvrhdn) [reply docs/tempo/website/configuration/manifest.md:21]: Done :+1:

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**